content/articles/2019-06-26-ragg-0-1-0.Rmarkdown

```diff
@@ -0,0 +1,109 @@
+---
+title: ragg 0.1.0
+author: Thomas Lin Pedersen
+date: '2019-06-26'
+slug: ragg-0-1-0
+description: >
+    A set of new graphic devices just landed on CRAN. Read more about the ragg package here.
+categories:
+  - package
+tags:
+  - r-lib
+photo:
+  url: https://unsplash.com/photos/OttsVAS4a84
+  author: Colin Watts
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE, comment = "#>", 
+  fig.width = 7, 
+  fig.align = 'center',
+  fig.asp = 0.618, # 1 / phi
+  out.width = "700px"
+)
+```
+
+We're stoked to announce the release of [ragg 0.1.0](https://ragg.r-lib.org) on CRAN. ragg is a package providing a set of high quality and high performance raster devices, capable of producing png, tiff, or ppm files as well as being available as a matrix of color values directly within R.
+
+ragg is part of our broader effort to improve graphics performance and quality in R at all levels of the stack, so that you'll benefit no matter what plotting framework you choose to use. Other parts of this efforts have been:
+
+- Developing the [devoid](https://github.com/r-lib/devoid) package to allow more precise benchmarking of plotting code.
+- Multiple improvements to rendering speed in grid in the latest R release (3.6.0).
+- Performance improvements in [ggplot2](https://ggplot2.tidyverse.org) 3.2.0 both broadly and for sf plotting specifically.
+- Performance improvements in [gtable](https://gtable.r-lib.com) 0.3.0
+
+## The devices
+In R a graphic device is an object that receives instructions from the graphic engine (such as draw a circle at this location) and translates that into some meaningful format for viewing, either in a file, on screen, or in some other form. ragg provides a set of raster devices, that is, devices that rasterize the instructions and writes it to some sort of raster output (e.g. a png file). This is opposed to vector devices such as `pdf()` and `svg()` that does not perform rasterization but writes the instructions directly into a vector graphics format.
```

Comment:
What's a graphics engine?

Reply:
The graphic engine in R (layer between grid/base plot and the device... should I expand on this?

Reply:
Yeah, I think so. It's just weird to define a term in terms of something else that people won't know about

Reply:
How about now?